src/hypothesis/searchstrategy/types.py

```diff
@@ -37,6 +37,38 @@ def type_sorting_key(t):
     return issubclass(t, collections.abc.Container)
 
 
+def resolve_forward_reference(thing):
+    """Resolve a forward reference, using typing internals."""
+    # First, we try the obvious: return immediately if it's not a ForwardRef
+    try:
+        from typing import _ForwardRef
+    except ImportError:  # pragma: no cover
+        return thing
+    if type(thing) != _ForwardRef:
```

Comment:
Is there a reason why this isn't `isinstance(thing, _ForwardRef)`? I think we have that lint turned off, but it still seems more natural.

Reply:
Alas, the authors of the `typing` module are *really* keen on metaclass magic, to the point of disabling instance and subclass checks for some generic types.  I don't remember now whether this was an actual problem or merely precautionary, but there's good reason for comparing to the type 😕